lib/compat/wordpress-6.1/block-editor-settings.php

```diff
@@ -13,6 +13,23 @@
  * @return array New block editor settings.
  */
 function gutenberg_get_block_editor_settings( $settings ) {
+	global $post_id;
+	$template_slug    = get_page_template_slug( $post_id );
+	$current_template = gutenberg_get_block_templates( array( 'slug__in' => array( $template_slug ) ) );
```

Comment:
A few surface notes:

* Since this is a post-editor-only setting, we might want to use the `block_editor_settings_all` filter and run logic only in the right editor `$context`.
* The [`get_page_template_slug`](https://developer.wordpress.org/reference/functions/get_page_template_slug/) only works for Page post types.

I think it would be nice to have a `getEditedPostTemplate` counterpart on the PHP side. Currently, we have no way to get an edited post-type template before the editor loads.

Reply:
> The [get_page_template_slug](https://developer.wordpress.org/reference/functions/get_page_template_slug/) only works for Page post types

It works for both posts and pages, though I haven't tried custom post types yet. The problem I'm encountering is that this function returns an empty string if the template in use is the default one, so we'll need some logic to check what post type it is and work out what its slug should be. I wish there were some sort of flag for this function to always return the slug, but it seems to have been written with a very specific purpose 🤷

Reply:
Sorry, you're right.

We can probably adopt logic from [`get_single_template`](https://developer.wordpress.org/reference/functions/get_single_template/) to cover all cases from the template hierarchy.

Here's my untested example:

```php
function wp_get_edited_post_template( $post_id ) {
	$object = get_post( $post_id );

	$templates = array();

	if ( ! empty( $object->post_type ) ) {
		$template = get_page_template_slug( $object );
		if ( $template && 0 === validate_file( $template ) ) {
			$templates[] = $template;
		}

		$name_decoded = urldecode( $object->post_name );
		if ( $name_decoded !== $object->post_name ) {
			$templates[] = "single-{$object->post_type}-{$name_decoded}";
		}

		$templates[] = "single-{$object->post_type}-{$object->post_name}";
		$templates[] = "single-{$object->post_type}";
	}

	$templates[] = 'single';

	return resolve_block_template( 'single', $templates, '' );
}
```

Reply:
So I'm not sure any solution will work for 100% of cases 😅, but I was looking into the return value of `get_default_block_template_types()` earlier and it looks like for most block themes we should expect 'single' to be the default template for posts, and 'page' to be the default for pages. But if those templates don't exist, we fall back to a 'singular' template that covers both posts and pages. I tried adding some logic to account for that and so far it's testing well on the handful of themes I looked at. 

I think the biggest problem here is that, until the user changes the template used for a specific post or page, we have no template data for it, so unless I'm mistaken we'll always have to infer it from the post type and the standard default template slugs.

Reply:
We'll need to calibrate the new function for our needs. This is a rough example based on the core function.

P.S. Starting from WP 6.1, user will be able to create templates for single entities - `"single-{$object->post_type}-{$object->post_name}"`.